### PR TITLE
feat: groups are collapsable in explorer sidebar even when a field is…

### PR DESCRIPTION
**Pull request by yasmin-e**
Closes: #4210 

### Description

In the Explore sidebar, when you select a field from the toggle list, the toggle list can no longer be collapsed. This is a bit annoying if you have really wide tables with loads of dimensions. 

Screenshot 2023-01-18 at 17 00 22

- [x] I can collapse the group even if there is a selected field in the group
- [x] I can see a tag with the number of selected fields near the group label when the group is **collapsed**

### CODE REVIEW

**Comment by hamzahc1** [comment]
Seeing something odd when I try to collapse a group with a selected dimension:

https://user-images.githubusercontent.com/14341285/214655272-cebe5194-bb49-4599-997d-074c7d21ab02.mov

**Review by hamzahc1 — CHANGES_REQUESTED** [review]
Seeing something odd when I try to collapse a group with a selected dimension. a negative number pops up?:

https://user-images.githubusercontent.com/14341285/214675197-fe5fdcd9-4843-4349-b933-959a5565ebdc.mov

File: packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeGroupNode.tsx [hunk 1]
```diff
@@ -26,17 +27,15 @@ const TreeGroupNode: FC<{ node: GroupNode; depth: number }> = ({
...
         hasIntersection(allChildrenKeys, Array.from(searchResults));
     const forceOpen = isSearching && hasVisibleChildren;
-    const isDisabled = hasSelectedChildren || forceOpen;
-
-    useEffect(() => {
-        if (hasSelectedChildren) {
-            toggle(true);
-        }
-    }, [hasSelectedChildren, toggle]);
+    const isDisabled = forceOpen;
```
Comment (owlas) [inline packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeGroupNode.tsx:38]: This variable isn't needed

File: packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeGroupNode.tsx [hunk 2]
```diff
@@ -26,17 +27,15 @@ const TreeGroupNode: FC<{ node: GroupNode; depth: number }> = ({
         allChildrenKeys,
         Array.from(selectedItems),
     );
+    const selectedChildrenCount = intersectionBy(
```
Comment (owlas) [inline packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeGroupNode.tsx:30]: How was this being calculated before?

**Review by owlas — CHANGES_REQUESTED** [review]

File: packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeGroupNode.tsx (hunk 2 above)
Comment (yasmin-e) [reply packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeGroupNode.tsx:30]: it wasn't

**Pull request merged**